File: 49-group-anagrams/group-anagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams_dict = defaultdict(list)
        
        for word in strs:
            sorted_word = ''.join(sorted(word))
            anagrams_dict[sorted_word].append(word)

        return list(anagrams_dict.values())

File: 49-group-anagrams/group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const anagramMap = strs.reduce((map, word) => {
        const key = word.split('').sort().join('');
        if (!map.has(key)) {
            map.set(key, [])
        }
        map.get(key).push(word);
        return map;
    }, new Map<string, string[]>());

    return Array.from(anagramMap.values());
}